contrib/endpoints/src/api_manager/config_manager.h

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef API_MANAGER_CONFIG_MANAGER_H_
+#define API_MANAGER_CONFIG_MANAGER_H_
+
+#include "contrib/endpoints/src/api_manager/context/global_context.h"
+#include "contrib/endpoints/src/api_manager/service_management_fetch.h"
+
+namespace google {
+namespace api_manager {
+
+namespace {
+
+// RolloutApplyFunction is the callback provided by ApiManager.
+// ConfigManager calls the callback after the service config download
+//
+// status
+//  - Code::OK        Config manager was successfully initialized
+//  - Code::ABORTED   Fatal error
+//  - Code::UNKNOWN   Config manager was not initialized yet
+// configs - pairs of ServiceConfig in text and rollout percentages
+typedef std::function<void(
+    const utils::Status& status,
+    const std::vector<std::pair<std::string, int>>& configs)>
+    RolloutApplyFunction;
+
+// Data structure to fetch configs from rollouts
+struct ConfigsFetchInfo {
```

Comment:
This is doable but private member FetchConfigs is using this data type as an argument.
To move this to config_manager.cc, FetchConfigs method should be moved outside of the ConfigManager class.